tests/bluetooth/host/buf/bt_buf_get_cmd_complete/src/main.c

```diff
@@ -0,0 +1,170 @@
+ *       and the same timeout value passed to bt_buf_get_cmd_complete()
+ *     - bt_dev.sent_cmd value isn't changed after calling bt_buf_get_cmd_complete()
+ *     - bt_buf_get_cmd_complete() returns NULL
+ *
+ */
+void test_returns_null_sent_cmd_is_null(void)
+{
+	struct net_buf *buf;
+	struct net_buf *before_call_sent_cmd_value;
+	k_timeout_t timeout = Z_TIMEOUT_TICKS(1000);
+
+	bt_dev.sent_cmd = NULL;
+	before_call_sent_cmd_value = bt_dev.sent_cmd;
+
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_evt_pool());
+#else
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_hci_rx_pool());
+#endif
+
+	ztest_returns_value(net_buf_alloc_fixed, NULL);
+	ztest_expect_value(hooks_net_buf_alloc_fixed_timeout_validation_hook, value, timeout.ticks);
+	buf = bt_buf_get_cmd_complete(timeout);
+	zassert_is_null(buf,
+			"bt_buf_get_cmd_complete() returned not NULL value while expecting NULL");
+	zassert_true((bt_dev.sent_cmd == before_call_sent_cmd_value),
+		     "bt_buf_get_type() caused bt_dev.sent_cmd value to be changed");
+}
+
+/*
+ *  Return value from bt_buf_get_cmd_complete() shouldn't be NULL
+ *
+ *  Constrains:
+ *     - bt_dev.sent_cmd value is NULL
+ *     - Timeout value is a positive non-zero value
+ *     - net_buf_alloc_fixed() return a not NULL value
+ *
+ *  Expected behaviour:
+ *     - net_buf_alloc_fixed() to be called with the correct memory allocation pool
+ *       and the same timeout value passed to bt_buf_get_cmd_complete()
+ *     - bt_dev.sent_cmd value isn't changed after calling bt_buf_get_cmd_complete()
+ *     - bt_buf_get_cmd_complete() returns the same value returned by net_buf_alloc_fixed()
+ *     - Return buffer type is set to BT_BUF_EVT
+ *
+ */
+void test_returns_not_null_sent_cmd_is_null(void)
+{
+	static struct net_buf test_reference;
+	struct net_buf *buf;
+	struct net_buf *before_call_sent_cmd_value;
+	k_timeout_t timeout = Z_TIMEOUT_TICKS(1000);
+
+	bt_dev.sent_cmd = NULL;
+	before_call_sent_cmd_value = bt_dev.sent_cmd;
+
+#if defined(CONFIG_BT_HCI_ACL_FLOW_CONTROL)
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_evt_pool());
+#else
+	ztest_expect_value(net_buf_alloc_fixed, pool, bt_buf_get_hci_rx_pool());
+#endif
+
+	ztest_expect_value(net_buf_simple_reserve, buf, &test_reference.b);
+	ztest_returns_value(net_buf_alloc_fixed, &test_reference);
+	ztest_expect_value(hooks_net_buf_alloc_fixed_timeout_validation_hook, value, timeout.ticks);
+	buf = bt_buf_get_cmd_complete(timeout);
+	zassert_equal(buf, &test_reference,
+		      "bt_buf_get_cmd_complete() returned incorrect buffer pointer value");
+
+	uint8_t type;
+
+	type = ((struct bt_buf_data *)net_buf_user_data(buf))->type;
+	zassert_equal(type, BT_BUF_EVT,
+		      "bt_buf_get_cmd_complete() returned incorrect buffer type %u, expected %u (%s)",
+		      bt_buf_get_type(buf), BT_BUF_EVT, STRINGIFY(BT_BUF_EVT));
+	zassert_equal(bt_buf_get_type(buf), type,
```

Comment:
This inline function is so simple that it can be checked while testing others APIs.
Creating a dedicated project for it won't add a value because it will require calling other functions (e.g. bt_buf_get_rx()) to set the type and then check that it was set correctly which is the same logic here.